vscode-dotnet-runtime-library/src/Acquisition/DirectoryProviderFactory.ts

```diff
@@ -12,5 +13,8 @@ import { SdkInstallationDirectoryProvider } from './SdkInstallationDirectoryProv
 
 export function directoryProviderFactory(mode: DotnetInstallMode, storagePath: string) : IInstallationDirectoryProvider
 {
-    return mode === 'runtime' ? new RuntimeInstallationDirectoryProvider(storagePath) : new SdkInstallationDirectoryProvider(storagePath);
+    return mode === 'runtime' ? new RuntimeInstallationDirectoryProvider(storagePath) :
+    mode === 'sdk' ? new SdkInstallationDirectoryProvider(storagePath) :
+    mode === 'aspnetcore' ? new ASPNetRuntimeInstallationDirectoryProvider(storagePath)
+    : new RuntimeInstallationDirectoryProvider(storagePath); // default if no mode is provided - should never happen
```

Comment:
If it should never happen, should this throw an error or fail in some other way?

Reply:
Hm, that's a good point -- it could! I think it's a matter of preference? In the future if someone adds a new mode and forgets to update this it will fail, which could be good because it will let them know they should update it.

That said, even if they dont update it, it will function properly. Its also pretty clear IMO that this is an area that can be updated when a new mode is added, so I lean towards keeping it. This has also been vendor tested so I am more hesitant to take potential changes like this one, I might do it in another PR.